### DISCUSSION TITLE
Preferable way of using FormRequest

### DISCUSSION THREAD

**Discussion by jappi00** [filamentphp/filament#6667] [Help] [answered]
Hello together,

My current project will provide an admin interface and an api. For that I want to use the same validation rules which I defined in an FormRequest:
```php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;
use Milwad\LaravelValidate\Rules\ValidDomain;

class ServerRequest extends FormRequest
{
    public function rules(): array
    {
        return [
            'name' => ['required', new ValidDomain()],
            'provider' => ['required'],
            'type' => ['required'],
            'token' => ['required'],
            'ipv4' => ['required'],
            'ipv6' => ['required'],
            'root_password' => ['required'],
            'image' => ['required'],
        ];
    }
}
```

Is there a best practice approach how I can implement this in the filament admin?

**Comment by danharrin** [comment, marked as answer]
There isn't a good way to use this. It doesn't follow our patten of storing the validation on the field-level. We provide some frontend validation as well as backend validation, so IMO you're losing out if you don't use our validation methods.

**Reply by jappi00** [reply]
Thank you very much. Than I have to duplicate the complete validation.
